diagnostics/cmd.py

```diff
@@ -19,3 +20,14 @@ def run_cmd(cmd):
     )
     stdout, stderr = proc.communicate()
     return stdout, stderr
+
+
+def run_root_cmd(cmd, passwd='kano'):
+    with open(os.devnull, 'w') as null_f:
+        proc = subprocess.Popen(
+            ['sudo', '-S', '-k'] + shlex.split(cmd),
+            stdin=subprocess.PIPE,
+            stdout=null_f,
+            stderr=null_f
+        )
+        proc.communicate('{}\n'.format(passwd))
```

Comment:
What if the user has changed their password?
I thought this whole script would have to be run with sudo.  Maybe you can detect that case and prompt them to do so, but then you have to test it twice.